File: BungieNetApi/Models/DestinyProgressionMappingDefinition.cs
using System.Runtime.Serialization;

namespace GhostSharp.BungieNetApi.Models
{
	/// <summary>
	/// Aggregations of multiple progressions.
	/// </summary>
	/// <summary>
	/// These are used to apply rewards to multiple progressions at once. They can sometimes have human readable data as well, but only extremely sporadically.
	/// </summary>
	[DataContract]
	public class DestinyProgressionMappingDefinition{

		/// <summary>
		/// Infrequently defined in practice. Defer to the individual progressions' display properties.
		/// </summary>
		[DataMember(Name="displayProperties", EmitDefaultValue=false)]
		public DestinyDisplayPropertiesDefinition DisplayProperties { get; set; }

		/// <summary>
		/// The localized unit of measurement for progression across the progressions defined in this mapping. Unfortunately, this is very infrequently defined. Defer to the individual progressions' display units.
		/// </summary>
		[DataMember(Name="displayUnits", EmitDefaultValue=false)]
		public string DisplayUnits { get; set; }

		/// <summary>
		/// The unique identifier for this entity. Guaranteed to be unique for the type of entity, but not globally.
		/// </summary>
		/// <summary>
		/// When entities refer to each other in Destiny content, it is this hash that they are referring to.
		/// </summary>
		[DataMember(Name="hash", EmitDefaultValue=false)]
		public uint Hash { get; set; }

		/// <summary>
		/// The index of the entity as it was found in the investment tables.
		/// </summary>
		[DataMember(Name="index", EmitDefaultValue=false)]
		public long Index { get; set; }

		/// <summary>
		/// If this is true, then there is an entity with this identifier/type combination, but BNet is not yet allowed to show it. Sorry!
		/// </summary>
		[DataMember(Name="redacted", EmitDefaultValue=false)]
		public bool Redacted { get; set; }


		public override bool Equals(object input)
        {
            return this.Equals(input as DestinyProgressionMappingDefinition);
        }

		public bool Equals(DestinyProgressionMappingDefinition input)
		{
			if (input == null) return false;

			return
				(
                    DisplayProperties == input.DisplayProperties ||
                    (DisplayProperties != null && DisplayProperties.Equals(input.DisplayProperties))
                ) &&
				(
                    DisplayUnits == input.DisplayUnits ||
                    (DisplayUnits != null && DisplayUnits.Equals(input.DisplayUnits))
                ) &&
				(
                    Hash == input.Hash ||
                    (Hash != null && Hash.Equals(input.Hash))
                ) &&
				(
                    Index == input.Index ||
                    (Index != null && Index.Equals(input.Index))
                ) &&
				(
                    Redacted == input.Redacted ||
                    (Redacted != null && Redacted.Equals(input.Redacted))
                ) ;
		}
	}
}

File: pkg-generated/models/DestinyProgressionMappingDefinition.go
package bungieAPI

// Aggregations of multiple progressions.
// These are used to apply rewards to multiple progressions at once. They can sometimes have human
// readable data as well, but only extremely sporadically.
type DestinyProgressionMappingDefinition struct {

	// Infrequently defined in practice. Defer to the individual progressions' display
	// properties.
	DisplayProperties DestinyDisplayPropertiesDefinition `json:"displayProperties"`

	// The localized unit of measurement for progression across the progressions defined in this
	// mapping. Unfortunately, this is very infrequently defined. Defer to the individual
	// progressions' display units.
	DisplayUnits string `json:"displayUnits"`

	// The unique identifier for this entity. Guaranteed to be unique for the type of entity, but not
	// globally.
	// When entities refer to each other in Destiny content, it is this hash that they are referring to.
	Hash int `json:"hash"`

	// The index of the entity as it was found in the investment tables.
	Index int `json:"index"`

	// If this is true, then there is an entity with this identifier/type combination, but BNet is not
	// yet allowed to show it. Sorry!
	Redacted bool `json:"redacted"`
}
